src/_ert_job_runner/client.py

```diff
@@ -74,18 +86,32 @@ async def _send(self, msg: AnyStr) -> None:
                 await self.websocket.send(msg)
                 return
             except ConnectionClosedOK:
-                pass  # Connection was closed no point in trying to send more messages
-                raise
+                logger.debug("Connection was closed from the server with ok message!")
+                return
             except (
-                ConnectionClosedError,
+                InvalidHandshake,
+                InvalidURI,
+                asyncio.TimeoutError,
                 ConnectionRefusedError,
                 OSError,
-                asyncio.TimeoutError,
-            ):
+            ) as exception:
+                if retry == self._max_retries:
+                    _error_msg = (
+                        f"Not able to establish the "
+                        f"websocket connection {self.url}! Max retries reached!"
+                        f" Exception from {type(exception)}: {str(exception)}"
+                    )
+                    raise ClientConnectionError(_error_msg) from exception
+            except ConnectionClosedError as exception:
                 if retry == self._max_retries:
-                    raise
-                await asyncio.sleep(0.2 + self._timeout_multiplier * retry)
-                self.websocket = None
+                    _error_msg = (
+                        f"Not been able to send the event"
+                        f" to {self.url}! Max retries reached!"
+                        f" Exception from {type(exception)}: {str(exception)}"
+                    )
+                    raise ClientConnectionError(_error_msg) from exception
+            await asyncio.sleep(0.2 + self._timeout_multiplier * retry)
+            self.websocket = None
```

Comment:
Is this indentation change deliberate or by accident? By reading the code it might be that the indent level of this might not matter though due to the `return`.

Reply:
This indentation is deliberate as it is valid for both exceptions when we don't raise the exception. We might put a new line in between though.